File: spring/member-server/src/main/java/com/example/memberserver/infrastructure/IdProvier/firebase/FirebaseConfiguration.java
package com.example.memberserver.infrastructure.IdProvier.firebase;

import com.google.auth.oauth2.GoogleCredentials;
import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.io.Resource;

import javax.annotation.PostConstruct;
import java.io.IOException;

@Configuration
public class FirebaseConfiguration {

    @Value("classpath:firebase-service-account.json")
    private Resource serviceAccount;

    @PostConstruct
    public void init() {
        try {
            FirebaseOptions options = FirebaseOptions.builder()
                    .setCredentials(GoogleCredentials.fromStream(serviceAccount.getInputStream()))
                    .setConnectTimeout(5000)
                    .setReadTimeout(5000)
                    .build();

            if (FirebaseApp.getApps().isEmpty()) {
                FirebaseApp.initializeApp(options);
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: spring/member-server/src/main/kotlin/com/example/memberserver/infrastructure/firebase/FirebaseConfiguration.kt
package com.example.memberserver.infrastructure.firebase

import com.google.auth.oauth2.GoogleCredentials
import com.google.firebase.FirebaseApp
import com.google.firebase.FirebaseOptions
import javax.annotation.PostConstruct
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration
import org.springframework.core.io.Resource

@Configuration
class FirebaseConfiguration {

    @Value("classpath:firebase-service-account.json")
    var serviceAccount: Resource? = null

    @PostConstruct
    fun init() {
        val options = FirebaseOptions.builder()
            .setCredentials(GoogleCredentials.fromStream(serviceAccount!!.inputStream))
            .setConnectTimeout(5000)
            .setReadTimeout(5000)
            .build()
        if (FirebaseApp.getApps().isEmpty()) {
            FirebaseApp.initializeApp(options)
        }
    }
}
